### ISSUE TITLE
Kalender mit Events & Community Dates

### ISSUE THREAD

**Issue by ffalt** [okfde/old.okfn.de#65]
Wunsch von Mara: 
Kalender wie bei http://codefor.de/termine/ aber automatisch aus dem OKF Public Kalender

**Comment by arnese** [comment]
Das würde ich gerne wieder aufgreifen. 
Muss das als externer Dienst laufen? Wäre eine kontinuierliche Pflege zu aufwändig? 
Zumindest beinhalten sollte ein Termin:

- Datum
- Art der Veranstaltung und Titel
- Person/Projekt
- Ort

**Comment by MaryP** [comment]
Danke @arnese ! Ich finde das immer noch eine gute Idee. Wir haben so viele events zu denen wir gehen und ich werde öfter gefragt wo seid ihr demnächst  - gibt es eine einfache Möglichkeit das zu basteln?

**Comment by arnese** [comment]
Wenn man schon dabei wäre, könnte man auch Presseartikel über OKF dazu nehmen

**Comment by ffalt** [comment]
ich seh da 3 Möglichkeiten:

**Comment by ffalt** [comment]
1. per Hand gepflegte yaml-Datei
2. OKF Public Kalender -> Export .ics -> Jekyll
3. Nur OKF Public Kalender als Iframe

zu 1: Wir haben volle Freiheit über Felder und Aussehen, aber die yaml-Datei muss regelmäßig gepflegt werden
zu 2: Wir haben volle Freiheit Aussehen, Felder sind durch Google Kalender beschränkt, lässt sich automatisieren, lässt sich an zentraler Stelle editieren
zu 3: Es sieht aus wie Google das will, lässt sich an zentraler Stelle editieren

Beispiel: https://okfn.de/termine/

**Comment by seegerseeger** [comment]
Ich finde 2 am besten insofern das technisch nicht zu aufwändig ist. Dann müssen wir nur beim Termine eintragen drauf achten, dass es nicht aussieht wie Kraut und Rüben, sollte doch zu schaffen sein ;)

**Comment by arnese** [comment]
Ist das Beispiel jetzt aus 2) oder sähe das in jedem Fall so aus? Sieht auf jeden Fall gut aus!

1 ist, glaube ich, zu aufwändig, als dass es von allen gemacht werden würde
2 scheint mir auch die einfachste Möglichkeit zu sein; wenn es dann fertig steht zumindest
3 wär nicht so cool, weil offensichtlich Google, glaube ich

**Comment by ffalt** [comment]
Im Beispiel ist oben: 2) ein einsamer Eintrag als 'Import aus .ics-Datei' Proof-of-concept und unten: 3) der Google IFrame Kalender

Ich bin auch für 2) als guter Kompromiss zwischen Pflegeaufwand und Designfreiheit und schaue mal wie sich die Automatisierung bewerkstelligen lässt.

**Comment by mattfullerton** [comment]
Ich wollte eine Frontend JS Lösung was die Daten live aus dem Google API hollt vorschlagen. Scheint technisch machbar aber ich finde kein fertiges Projekt was man integrieren könnte, also ich schätze zu viel Arbeit.

**Comment by ffalt** [comment]
dacht ich auch erst, aber für das api braucht man einen oauth key (soweit ich verstanden habe auch für public kalender) und der key im client-code ist ungünstig.
es gibt aber eine öffentliche [.ics datei ](
https://www.google.com/calendar/ical/okfn.de_jcb2bir16010snrta90clnthrg@group.calendar.google.com/public/basic.ics) für den kalender, d.h. regelmäßiger cron job auf dem server sollte es tun und ist wenig Arbeit (mit wget die datei herunterladen und jekyll neustarten). 
den ics-reader als plugin für jekyll habe ich schon, wir müssen die einträge nur noch schön darstellen (und gegenenfalls ein wenig javascript für bessere usability)

**Comment by mattfullerton** [comment]
Ja, ist eine gute Lösung, ich denke nur an den Denkkonflikt zwischen Kalendar kann sich jederzeit ändern, Jekyll nur periodisch. Aber ich habe das gleiche Thema mit 2030watch (nur da mit Sheets).

Das mit OAuth stimmt. Aber mit einer Application/User Key hätte man einen Key der wirklich nichts anders kann als von dem Kalendar lesen. Du gibst als OKF dieses Quasinutzer nur Leserecht auf dem *öffentlichen* Kalendar.

**Comment by ffalt** [comment]
ah danke für die info, ich wusste nicht dass sich die berechtigungen für keys unterschiedlich festlegen lassen.
über das Google API lassen sich auch webhooks einrichten, die dann bei änderung im kalender jekyll auslösen können - genauso wie wir es auch mit github machen. aber ich glaube eine aktualisierung pro stunde wird ausreichen. zuerst muss der kalender ohnehin überhaupt benutzt werden :)

**Comment by ffalt** [comment]
ich habe jetzt das styling von code-for adaptiert und das einlesen des google .ics verbessert [https://okfn.de/termine/](https://okfn.de/termine/) 

aber nun fällt mir auf die füße dass wir die felder nicht selbst bestimmen können. 
Ich habe im google calender kein z.B. standard-feld für eine URL gefunden, um einen link für einen termin einzutragen, (sodass diese in die .ics datei exportiert wird, denn das format kennt ein URL-feld). Das heisst, wir können nicht ohne weiteres einen "Mehr Info"-link anbieten, was schon recht wichtig wär. Mir fällt nur ein, URLs ins feld "Beschreibung" einzutragen und mit jeykll wieder rauszuholen, aber das ist eher unschön und kann mühsam zu programmieren sein (zumindest mit jekylls liquid-skriptsprache).

**Comment by mattfullerton** [comment]
Wie viel von der Beschreibung hattest du vor anzuzeigen? Ich würde Kalendareintragenden lieber die Empfehlung geben, schau dass eine Link mit mehr Infos in der Beschreibung drin ist als 'bitte in der ersten Zeil einen URL eintragen'. Und ja. parsen mit Jekyll macht wirklich kein spaß[1].

[1] :cry: https://github.com/okfde/2030-watch.de/blob/dev/_includes/visualizations/main_visualization.html#L34

**Comment by ffalt** [comment]
ich dachte an 
"kurze beschreibung"
"wer in der okf"
"typ der veranstaltung"
"wo (mit geolocation)" und 
"mehr informationen-link"

zumindest hatte ich mir das so vorgestellt weil es dafür felder im ics gibt, die google aber nicht füllt. da muss ich wohl streichen. und ja, du hast völlig recht, informale regelungen wie "url in 1. zeile" funktionieren nicht auf dauer.
